go/types/opcache.go

```diff
@@ -27,25 +45,65 @@ func newOpCache(vrw ValueReadWriter) *opCache {
 		WriteBuffer:            1 << 27, // 128MiB
 	})
 	d.Chk.NoError(err, "opening put cache in %s", dir)
-	return &opCache{ops: db, dbDir: dir, vrw: vrw}
+	return &opCacheDb{ops: db, dbDir: dir}
+}
+
+func newOpCache(vrw ValueReadWriter) *opCache {
+	incrementActiveColCount()
+	prefix := [prefixByteSize]byte{}
+	colId := atomic.AddUint32(&collectionId, 1)
```

Comment:
I feel like you don't need this atomic op here. You could jut increment `collectionId` in `incrementActiveColCount()` and return it, no? It just seems nice to reduce the number of synchronization primitives you need.